spk/octoprint/Makefile

```diff
@@ -1,40 +1,37 @@
 SPK_NAME   = octoprint
-SPK_VERS   = $(shell date +%Y%m%d)
-SPK_REV    = 4
+SPK_VERS   = 1.4.2
+SPK_REV    = 5
 SPK_ICON   = src/octoprint.png
 DSM_UI_DIR = app
 
-BETA = 1
-
-BUILD_DEPENDS = cross/python cross/setuptools cross/pip cross/wheel
+BUILD_DEPENDS = cross/python3 cross/setuptools cross/pip cross/wheel
 DEPENDS = cross/$(SPK_NAME)
 WHEELS  = src/requirements.txt
+PIP = pip3
 
-SPK_DEPENDS = "python>=2.7.11-15"
+SPK_DEPENDS = "python3>=3.7"
 
 MAINTAINER   = SynoCommunity
-DESCRIPTION  = OctoPrint is a so called host software for 3D printers that controls your 3D printer and sends it the actual commands to do its job.
+DESCRIPTION  = OctoPrint provides a snappy web interface for controlling consumer 3D printers.
 ADMIN_PORT   = 8088
 RELOAD_UI    = yes
 DISPLAY_NAME = OctoPrint
-HOMEPAGE     = http://octoprint.org/
+HOMEPAGE     = https://octoprint.org/
 LICENSE      = AGPLv3
 
 INSTALLER_SCRIPT = src/installer.sh
 SSS_SCRIPT       = src/dsm-control.sh
```

Comment:
Is there a reason not to migrate to generic service support and common installer code?

Reply:
I will try this, but the start and stop functions are very specific here.

Reply:
Then I will give it a try myself